### PR TITLE
PLT-117 - fixed conditional, added support for displayName upon acces…

### PR DESCRIPTION
**Pull request by yoavdr**
- fixed conditional
- added support for displayName upon access policies creation and update
- added patterns to .gitignore.

### CODE REVIEW

File: provider/resource_web_access_policy.go [hunk 1]
```diff
@@ -80,6 +80,21 @@ func resourceCreateWebAccessPolicy(d *schema.ResourceData, m interface{}) error
 			return errors.New(error)
 		}
 		accessPolicy.DirectoryEntities[i].IdentityProviderType = resolvedIdentityProviderType
+
+		// Get Display Name for User/Group by ID
+		var resolvedDisplayName string
+		switch accessPolicy.DirectoryEntities[i].EntityType {
```
Comment (orensh0) [inline provider/resource_web_access_policy.go:86]: accessPolicy.DirectoryEntities[i].EntityType to lower case
case against lower case

Comment (orensh0) [inline provider/resource_web_access_policy.go:86]: no default case if non expected EntityType?

File: service/dto/access_policy_converter.go [hunk 2]
```diff
@@ -122,10 +122,11 @@ func ConvertToDto(accessPolicy *AccessPolicy) sdk.PolicyAccess {
 
 	for _, directoryEntity := range accessPolicy.DirectoryEntities {
 		identityProviderType, err := ConvertIdentityProviderTypeToEnum(directoryEntity.IdentityProviderType)
-		if err != nil {
+		if err == nil {
 			directoryEntities = append(directoryEntities, sdk.DirectoryEntity{
 				IdentifierInProvider: directoryEntity.IdentifierInProvider,
 				IdentityProviderId:   directoryEntity.IdentityProviderId,
+				DisplayName: 		  directoryEntity.DisplayName,
```
Comment (orensh0) [inline service/dto/access_policy_converter.go:129]: missing go fmt?

File: service/dto/access_policy_converter.go [hunk 3]
```diff
@@ -122,10 +122,11 @@ func ConvertToDto(accessPolicy *AccessPolicy) sdk.PolicyAccess {
 
 	for _, directoryEntity := range accessPolicy.DirectoryEntities {
 		identityProviderType, err := ConvertIdentityProviderTypeToEnum(directoryEntity.IdentityProviderType)
-		if err != nil {
+		if err == nil {
```
Comment (orensh0) [inline service/dto/access_policy_converter.go:125]: add comment why we create it in case of error

File: service/dto/dto.go [hunk 4]
```diff
@@ -136,6 +136,7 @@ type DirectoryEntity struct {
 	IdentityProviderId   string
 	EntityType           string
 	IdentityProviderType string
+	DisplayName			 string
```
Comment (orensh0) [inline service/dto/dto.go:139]: missing fo fmt?

File: service/identity_provider.go [hunk 5]
```diff
@@ -45,4 +45,22 @@ func (u *IdentityProviderAPI) GetIdentityProviderTypeById(identityProviderId str
...
 	return "", errors.Errorf("can't find identity provider with id '%s'", identityProviderId)
+}
+
+func (u *IdentityProviderAPI) GetUserDisplayNameTypeById(identityProviderId string, IdentifierInProvider string) (string, error) {
+	user, _, err := u.cli.UsersApi.GetUser(context.Background(), identityProviderId, IdentifierInProvider)
+	if err != nil {
+		return "", err
+	}
+
+	return user.FirstName + " " + user.LastName, nil
```
Comment (orensh0) [inline service/identity_provider.go:56]: could use Sprintf instead of creating 3 strings

File: service/dto/access_policy_converter.go (hunk 3 above)
Comment (yoavdr) [reply service/dto/access_policy_converter.go:125]: We only create if if there is no error:

if err == nil {
			directoryEntities = append(directoryEntities, sdk.DirectoryEntity{
                        ...
                        ...
}

**Pull request merged**